model_analyzer/perf_analyzer/perf_analyzer.py

```diff
@@ -68,18 +74,30 @@ def run(self, tags):
 
         if tags:
             # Synchronously start and finish run
-            cmd = [self.bin_path]
-            cmd += self._config.to_cli_string().replace('=', ' ').split()
-            try:
-                self._output = check_output(cmd,
-                                            start_new_session=True,
-                                            stderr=STDOUT,
-                                            encoding='utf-8')
-            except CalledProcessError as e:
-                raise TritonModelAnalyzerException(
-                    f"Running perf_analyzer with {e.cmd} failed with exit status {e.returncode} : {e.output}"
-                )
-        return [metric(self._output) for metric in tags]
+            for _ in range(MAX_INTERVAL_CHANGES):
+                cmd = [self.bin_path]
+                cmd += self._config.to_cli_string().replace('=', ' ').split()
+
+                try:
+                    self._output = check_output(cmd,
+                                                start_new_session=True,
+                                                stderr=STDOUT,
+                                                encoding='utf-8')
+                    return [metric(self._output) for metric in tags]
+                except CalledProcessError as e:
+                    if e.output.find("Please use a larger time window.") > 0:
+                        self._config['measurement-interval'] += INTERVAL_DELTA
+                        logger.info(
+                            f"perf_analyzer's measurement window is too small, increased to {self._config['measurement-interval']} ms."
+                        )
+                    else:
+                        raise TritonModelAnalyzerException(
+                            f"Running perf_analyzer with {e.cmd} failed with exit status {e.returncode} : {e.output}"
+                        )
+
+            raise TritonModelAnalyzerException(
+                f"Ran perf_analyzer {MAX_INTERVAL_CHANGES} number of times, but no valid requests recorded in even max time interval of {self._config['measurement-interval']} "
```

Comment:
I think you can remove "even" from the error message. Can you please elaborate what is the issue here? I'm just curious what was the issue that you encountered.

Reply:
If the measurement interval for perf analyzer is too small (try 50ms for example), it will say that there were no requests received in the time interval and ask you to increase the measurement window. This just automates the process. Yes I will edit this error message.